### PR TITLE
feat: Tags for services and ad-hoc run

### PR DESCRIPTION
**Pull request by guikcd**
*Issue #, if available:*
Closes: #605, closes: #538

*Description of changes:*

Add support for tags:
* service: propagate tags from service during an update
* run-task: set custom tags

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by guikcd** [comment]
I encountered an issue when wanted to propagate tags from task definition for run-task because of https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/249#issuecomment-1538851215. The describe-task-definition can return tags, but it need to be merged to the task definition, which is not easy to do with cli or in amazon-ecs-render-task-definition (see https://github.com/aws-actions/amazon-ecs-render-task-definition/issues/317).

**Comment by guikcd** [comment]
@kg-aws ready to review

File: index.js [hunk 1]
```diff
@@ -395,7 +397,9 @@ async function run() {
     const desiredCount = parseInt((core.getInput('desired-count', {required: false})));
     const enableECSManagedTagsInput = core.getInput('enable-ecs-managed-tags', { required: false }) || 'false';
     const enableECSManagedTags = enableECSManagedTagsInput.toLowerCase() === 'true';
-   
+    const propagateTags = core.getInput('propagate-tags', { required: false }) || 'NONE';
+    const runTaskTags = JSON.parse(core.getInput('run-task-tags', { required: false }) || '[]');
```
Comment (kg-aws) [inline index.js:401]: We can read the config related to runTask in the `runTask` function. So `runTaskTags` config can be read inside the `runTask` function like other configs starting from line 42.

File: index.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const IGNORED_TASK_DEFINITION_ATTRIBUTES = [
 ];
 
 // Method to run a stand-alone task with desired inputs
-async function runTask(ecs, clusterName, taskDefArn, waitForMinutes, enableECSManagedTags) {
+async function runTask(ecs, clusterName, taskDefArn, waitForMinutes, enableECSManagedTags, runTaskTags) {
```
Comment (kg-aws) [inline index.js:24]: The `runTaskTags` parameter can be removed by the reading the config `runTaskTags` config inside this function.

**Review by kg-aws — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (guikcd) [reply index.js]: ✅

File: index.js (hunk 2 above)
Comment (guikcd) [reply index.js]: ✅

**Review by kg-aws — APPROVED** [review]
LGTM!

File: index.test.js [hunk 3]
```diff
@@ -1446,4 +1467,36 @@ describe('Deploy to ECS', () => {
         expect(core.setFailed).toHaveBeenNthCalledWith(1, 'Failed to register task definition in ECS: Could not parse');
         expect(core.setFailed).toHaveBeenNthCalledWith(2, 'Could not parse');
     });
+
+    test('propagate service tags from service and enable ecs managed tags', async () => {
```
Comment (amazreech) [inline index.test.js:1471]: Hi @guikcd, thanks for this PR! 
I believe we are missing unit tests that checks the `run-task-tags` functionality. Would it also be possible to add unit tests for the `run-task-tags` parameter/functionality as part of existing or as a new test.

Comment (guikcd) [reply index.test.js:1473]: I've modified the existing test `run task with options` to verify run-task-tags.

**Review by amazreech — APPROVED** [review]
Thanks for the PR! LGTM!

**Pull request merged**